core/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -1250,6 +1350,152 @@ public boolean move() {
         public boolean cannotRemain() {
             return isDecisionTaken() && canRemainDecision.type() == Type.NO;
         }
+
+        /**
+         * Gets the individual node-level decisions that went into making the final decision as represented by
+         * {@link #getFinalDecisionType()}.  The map that is returned has the node id as the key and a {@link WeightedDecision}.
+         */
+        @Nullable
+        public Map<String, WeightedDecision> getNodeDecisions() {
+            return nodeDecisions;
+        }
+    }
+
+    /**
+     * Represents a decision to move a started shard to form a more optimally balanced cluster.
+     */
+    public static final class RebalanceDecision extends RelocationDecision {
+        /** a constant representing no decision taken */
+        public static final RebalanceDecision NOT_TAKEN = new RebalanceDecision(null, null, null, null, null, Float.POSITIVE_INFINITY);
```

Comment:
I think it would be nice if it added an explanation that explained it was out of the realm of responsibility, but I'm +0 on adding it, so whatever you'd prefer.